app/grandchallenge/archives/models.py

```diff
@@ -13,4 +14,9 @@ class Archive(UUIDModel):
     images = models.ManyToManyField(Image)
 
     def __str__(self):
-        return "<{} {}>".format(self.__class__.__name__, self.name)
+        return f"<{self.__class__.__name__} {self.name}>"
+
+    def delete(self, *args, **kwargs):
+        # Remove all related patients and other models via cascading
+        Patient.objects.filter(study__image__archive__id=self.id).delete()
```

Comment:
I think that .delete() should also get the `*args` and `**kwargs`.

What happens if the `Image` also belongs to another archive?

Reply:
It would have been deleted. I changed the method now.